From: Modestas Vainius <modestas@vainius.eu>
Subject: [PATCH] debian/aboutdata_add_debian_version

Add Debian package version to the About dialog and KDE
bug creation tool.

Signed-off-by: Modestas Vainius <modestas@vainius.eu>

---
 src/CMakeLists.txt |    5 +++++
 src/main.cpp       |   10 +++++++++-
 2 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e242545..0500f69 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -194,6 +194,11 @@ set(blowfish_SRCS
 set (completed_SRCS ${core_SRCS}  ${irc_SRCS} ${viewer_SRCS} ${prefs_SRCS} ${dcc_SRCS} ${linkaddressbook_SRCS} ${blowfish_SRCS})
 set (konversation_SRCS ${completed_SRCS})
 
+if (DEBIAN_PACKAGE_VERSION)
+    MESSAGE(STATUS "Building Debian package version ${DEBIAN_PACKAGE_VERSION} ...")
+    set_source_files_properties(main.cpp PROPERTIES COMPILE_FLAGS -DDEBIAN_PACKAGE_VERSION=${DEBIAN_PACKAGE_VERSION})
+endif (DEBIAN_PACKAGE_VERSION)
+
 kde4_add_ui_files(konversation_SRCS
     identitydialog.ui
     queuetunerbase.ui
diff --git a/src/main.cpp b/src/main.cpp
index b04e03e..27c4c8a 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -25,6 +25,14 @@
 #define HACKSTR(x) #x
 #define STRHACK(x) HACKSTR(x)
 
+#ifdef DEBIAN_PACKAGE_VERSION
+#  define _STRINGIFY(s) #s
+#  define _DEBIAN_APP_VERSION(debver) "(Debian " _STRINGIFY(debver) ")"
+#  define FULL_APP_VERSION(ver) ver " " _DEBIAN_APP_VERSION(DEBIAN_PACKAGE_VERSION)
+#else
+#  define FULL_APP_VERSION(ver) ver
+#endif
+
 /*
   Don't use i18n() here, use ki18n() instead!
   i18n() will only work as soon as a kapplication object was made.
@@ -35,7 +43,7 @@ int main(int argc, char* argv[])
     KAboutData aboutData("konversation",
         "",
         ki18n("Konversation"),
-        KONVI_VERSION " #" STRHACK(COMMIT),
+        FULL_APP_VERSION(KONVI_VERSION " #" STRHACK(COMMIT)),
         ki18n("A user friendly IRC client"),
         KAboutData::License_GPL,
         ki18n("(C) 2002-2009 by the Konversation team"),
-- 
tg: (62bc25a..) debian/aboutdata_add_debian_version (depends on: upstream)
